docs/design/2022-06-15-phyical_plan_explain.md

```diff
@@ -0,0 +1,603 @@
+# TiSpark Design Documents
+
+- Author(s): [qidi1](https://github.com/qidi1)
+- Tracking Issue: https://github.com/pingcap/tispark/issues/2385
+
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Motivation or Background](#motivation-or-background)
+- [Detailed Design](#detailed-design)
+- [Test Design](#test-design)
+
+## Introduction
+
+The physical plan is the physical execution plan in TiSpark. When we use `explain` in Spark which runs with TiSpark, The process of the physical plan will be displayed in the terminal. but now this display about how the plan executes has some problems.
+
+- outputting multiple filters which not should show together at the same time.
+
+- unclear representation of the execution process.
+
+- obscure push-down conditions.
+
+The display of the  physical plan needs to be improved.
+
+## Background
+
+### The Process of Scan
+
+#### IndexScan
+
+In TiSpark, An IndexScan requires two scans. One is scanning in index data to get the RowID and the other one is scanning in table data to get the real data. The first scan in code is called IndexScan, and the second scan in code called TableScan, but to facilitate the distinction, in the latter part of the text, the first scan is called IndexRangeScan, and the second scan is called TableRowIDScan.
+
+#### CoveringIndexScan
+
+CoveringIndexScan is a special case of IndexScan. If the column in Filter and the column in Projection are both inside the Index in one visit, then we only need to scan the Index once, no need to scan for the Table. that is, it only scans for the Index, such a Scan is called CoveringIndexScan.
+
+#### TableScan
+
+TableScan is different from IndexScan and CoveringIndexScan. TableScan only scans table data. such a scan we called TableRangeScan in the following.
+
+### Filter Concept
+
+#### PushDown Filter
+
+The expression passed to COP/TiKV as Selection without triggering a downgrade.
+
+#### Downgrade Filter
+
+Downgrade Filter is used after IndexScan downgrade.
```

Comment:
Can we give a short description about what downgrade causes? Millions of separated ranges will become a simple, large, continious range?

Reply:
add the description of what happened after trigger downgrade.